sonar/modules/projects/jsonresolvers.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Swiss Open Access Repository
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Project resolver."""
+
+from __future__ import absolute_import, print_function
+
+import jsonresolver
+from invenio_pidstore.resolver import Resolver
+from invenio_records.api import Record
+
+
+# the host corresponds to the config value for the key JSONSCHEMAS_HOST
+@jsonresolver.route('/api/projects/<pid>', host='sonar.ch')
+def project_resolver(pid):
+    """Resolve referenced project."""
+    resolver = Resolver(pid_type='proj', object_type="rec",
+                        getter=Record.get_record)
+    _, record = resolver.resolve(pid)
+
+    if record.get('$schema'):
+        del record['$schema']
+
+    return record
```

Comment:
Seems not the right approach. Changing the project name implies the reindex of all related documents.

Reply:
I will do that in another PR..